packages/editor/src/components/provider/use-block-editor-settings.js

```diff
@@ -125,6 +148,9 @@ function useBlockEditorSettings( settings, hasTemplate ) {
 			canUseUnfilteredHTML,
 			undo,
 			hasTemplate,
+
+			userCanCreatePages,
+			createPageEntity,
```

Comment:
This PR introduced a really bad degradation in terms of typing performance. I think this change here has been reverted though in a more recent PR so the degradation is gone but we should be careful with block editor settings. It shouldn't become a basket for any config. https://codehealth.vercel.app

Reply:
From eyeballing, it seems like the problem rooted in the fact that `createPageEntity` was a different function every time this function ran, therefore `useMemo` was re-ran on every re-render and possibly triggered another re-render.

Reply:
[Yes that's right](https://github.com/WordPress/gutenberg/pull/35083#discussion_r723878469).

> Specifically the dependencies we introduced to the useMemo hook that caused the entire memoized object to be recomputed on each render because the references were always different.